src/site/_includes/components/cookie/cookie-banner.ts

```diff
@@ -0,0 +1,74 @@
+import { analyticsSetting, removeAnalyticCookies } from './analytics'
+import { COOKIE_PREFERENCE_KEY, getCookie, setCookie } from './cookie'
+
+const COOKIE_BANNER_CLASS_PREFIX = 'js-cookie-guidance-banner'
+
+function getClassName(className: string) {
+  return `.${COOKIE_BANNER_CLASS_PREFIX}--${className}`
+}
+
+function getCookieMessageElements(
+  parentElement: Element,
+  types: string[],
+) {
+  return Object.fromEntries(types.map((type) => {
+    const element = parentElement.querySelector<HTMLDivElement>(getClassName(type))
+    if (!element) {
+      throw new Error('Message not found')
+    }
+    return [type, {
+      show: () => {
+        element.removeAttribute('hidden')
+      },
+      hide: () => {
+        element.setAttribute('hidden', 'hidden')
+      },
+    }]
+  }))
+}
+
+export function initCookieBanner() {
+  const cookiePromptEnabled = new URLSearchParams(document.location.search).has('cookie_prompt_enabled')
+  // Don't run on the cookies setting page
+  const cookieSettingForm = document.querySelector<HTMLFormElement>('.js-cookie-setting-form')
+  const cookieBannerEl = document.querySelector(getClassName('container'))
+
+  const cookiePreference = getCookie(COOKIE_PREFERENCE_KEY)
+
+  if (
+    !cookiePromptEnabled || cookiePreference === '1' || cookiePreference === '0' || !cookieBannerEl || cookieSettingForm
+  ) {
```

Comment:
Added a check here not to run the cookie banner on the cookies setting page similar to gov.uk